resources/sts/4.12/hypershift/InstallerPolicy.json

```diff
@@ -4,67 +4,130 @@
-                "iam:AddRoleToInstanceProfile",
-                "iam:AttachRolePolicy",
-                "iam:CreateInstanceProfile",
-                "iam:CreateOpenIDConnectProvider",
-                "iam:CreatePolicy",
-                "iam:CreatePolicyVersion",
-                "iam:CreateRole",
-                "iam:DeleteOpenIDConnectProvider",
-                "iam:DeletePolicyVersion",
-                "iam:DeleteRole",
-                "iam:DeleteRolePermissionsBoundary",
-                "iam:DetachRolePolicy",
-                "iam:GetOpenIDConnectProvider",
-                "iam:GetPolicy",
-                "iam:GetRole",
-                "iam:GetUser",
-                "iam:ListAttachedRolePolicies",
-                "iam:ListPolicyTags",
-                "iam:ListPolicyVersions",
-                "iam:ListRoleTags",
-                "iam:PutRolePermissionsBoundary",
-                "iam:TagPolicy",
                 "kms:DescribeKey",
-                "route53:ChangeResourceRecordSets",
-                "route53:DeleteHostedZone",
-                "route53:GetHostedZone",
+                "iam:GetRole",
                 "route53:ListHostedZones",
-                "route53:ListHostedZonesByName",
                 "route53:ListResourceRecordSets",
-                "s3:CreateBucket",
-                "s3:DeleteBucket",
-                "s3:DeleteObject",
-                "s3:ListObjects",
-                "s3:PutObject",
-                "secretsmanager:GetSecretValue",
                 "servicequotas:GetServiceQuota",
-                "sts:AssumeRole",
-                "sts:AssumeRoleWithWebIdentity",
                 "sts:GetCallerIdentity"
             ],
-        "Resource": "*"
+            "Resource": "*"
+        },
+        {
+            "Action": [
+                "iam:PassRole"
+            ],
+            "Resource": [
+                "arn:*:iam::*:role/*-Worker-Role"
+            ],
+            "Effect": "Allow"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+              "sts:AssumeRole",
+              "sts:AssumeRoleWithWebIdentity"
+            ],
+            "Resource": [
+              "arn:aws:iam::*:role/*"
+            ],
+            "Condition": {
+              "StringEquals": {
+                "aws:ResourceTag/red-hat-managed": "true"
+              }
+            }
+        },
+        {
+            "Effect": "Allow",
+            "Action": "elasticloadbalancing:CreateLoadBalancer",
+            "Resource": [
+              "*"
+            ],
+            "Condition": {
+              "StringEquals": {
+                "aws:RequestTag/red-hat-managed": "true"
+              }
+            }
+        },
+        {
+            "Sid": "BucketActions",
+            "Effect": "Allow",
+            "Action": [
+                "s3:CreateBucket",
+                "s3:DeleteBucket"
+            ],
+            "Resource": [
+                "arn:aws:s3:::*-oidc"
```

Comment:
Does HyperShift create a bucket in the customers account? I think we're moving towards BYO OIDC @arendej ?

Reply:
Hypershift is only BYO OIDC.
cc @wgordon17

Reply:
if hyperShift installs are going to enforce BYO-OIDC, then this permission would not be needed. I tried an install with BYO-OIDC configuration withouth these permissions and the cluster installed successfully.

I have removed these permission from the installer role now, please let us know if BYO-OIDC will not be enforced, in which case we will need to add these permissions back.

Reply:
@gdbranco could you share the adjusted BYO-OIDC flow? It's still possible that we might need S3 permissions, even with BYO-OIDC. It might not be _required_, since in BYO-OIDC customers can bring _whatever_, but I thought the default still relied on S3 ¯\_(ツ)_/¯